presto-jdbc/src/main/java/io/prestosql/jdbc/AbstractPrestoResultSet.java

```diff
@@ -613,7 +613,7 @@ public BigDecimal getBigDecimal(int columnIndex)
             return null;
         }
 
-        return new BigDecimal(String.valueOf(value));
+        return toBigDecimal(String.valueOf(value));
```

Comment:
if you change `toBigDecimal` to return Optional, this can become eg

```
return toBigDecimal(String.valueOf(value))
        .orElseThrow(() -> new SQLException("Value is not a number: " + value));
```

Reply:
fixed